lib/ansible/modules/packaging/os/redhat_repository.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/python
+
+# Giovanni Sciortino (giovannibattistasciortino@gmail.com)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: redhat_repository
+short_description: Manage RHSM repositories using the C(subscription-manager) command
+description:
+    - Manage(List/Enable/Disable) RHSM repositories to the Red Hat Subscription Management entitlement platform using the C(subscription-manager) command
+version_added: "2.4"
+author: "Giovanni Sciortino (giovannibattistasciortino@gmail.com)"
+notes:
+    - In order to manage rhsm repositories the system must be already registered to rhsm manually or using the ansible module redhat_subscription.
+requirements:
+    - subscription-manager
+options:
+    state:
+        description:
+          - If state is equal to 'enabled' or 'disabled', state indicates the desired repositories states.
+          - If state is equal to 'list', retries the list of repositories and store it in 'rhsm_repositories' return value
+        choices: [ "list", "enabled", "disabled" ]
+        required: True
+        default: None
+    name:
+        description:
+            - If state is equal to 'enabled' or 'disabled', id of repositories to enable.
+              To operate on several repositories this can accept a comma separated list or a list
+            - If state is equal to 'list', value of 'name' can be not specified(list all repositories) or must be one
+              of the following values all(list all repositories), enabled(list enabled repositories), disabled(list disabled repositories)
+        required: False
+        default: None
+'''
+
+EXAMPLES = '''
+- name: List all RHSM repositories.
+  redhat_repository:
+    state: list
+
+- name: List enabled RHSM repositories.
+  redhat_repository:
+    name: enabled
+    state: list
+
+- name: Enable a RHSM repository
+  redhat_repository:
+    name: rhel-7-server-rpms
+    state: enabled
+
+- name: Disable all RHSM repositories
+  redhat_repository:
+    name: '*'
+    state: disabled
+
+- name: Enable all repository starting with rhel-6-server
+  redhat_repository:
+    name: 'rhel-6-server*'
+    state: enabled
+
+'''
+
+RETURN = '''
+repositories:
+    description: if state is equal to list, rhsm_repositories is the list of repositories
+    returned: success
+    type: list
+'''
+
+from ansible.module_utils.basic import AnsibleModule
```

Comment:
I have seen that pep8 filename produces the warning "E402 module level import not at top of file". But if I move up the import Iabove the error, the test checked by shippable fails with the following error:
`lib/ansible/modules/packaging/os/redhat_repository.py:8:0: E106 Import found before documentation variables. All imports must appear below DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA.
`